openhcl/openhcl_boot/src/sidecar.rs

```diff
@@ -7,9 +7,10 @@ use crate::host_params::MAX_NUMA_NODES;
 use crate::host_params::PartitionInfo;
 use crate::host_params::shim_params::IsolationType;
 use crate::host_params::shim_params::ShimParams;
-use crate::single_threaded::off_stack;
-use arrayvec::ArrayVec;
-use memory_range::MemoryRange;
+use crate::memory::AddressSpaceManager;
+use crate::memory::AllocationPolicy;
+use crate::memory::AllocationType;
+use sidecar_defs::PAGE_SIZE;
```

Comment:
This is a good example of why I don't like the allocator taking a page size. Why would the _sidecar_ page size have anything to do with the allocator's page size? If sidecar were to use 64KB on ARM for some reason, would we need to use 64KB in the allocator to match?

Reply:
yeah and i _also_ fixed a bug here where i missed this size calculation. I'm inclined to roll it back to bytes and just round up to the nearest page size.